plaso/parsers/viminfo.py

```diff
@@ -0,0 +1,522 @@
+  _REGISTERS_ITEM = pyparsing.Group(
+      pyparsing.Suppress(pyparsing.LineStart()) +
+      pyparsing.Literal('"') +
+      pyparsing.Or([pyparsing.Word(pyparsing.nums),
+                    pyparsing.Word(pyparsing.printables)]) +
+      pyparsing.Suppress(pyparsing.White('\t')) +
+      pyparsing.Or(['CHAR', 'LINE', 'BLOCK']) +
+      pyparsing.Suppress(pyparsing.White('\t')) +
+      pyparsing.Word(pyparsing.nums) + pyparsing.Suppress(pyparsing.LineEnd()) +
+      pyparsing.Group(pyparsing.ZeroOrMore(_REGISTERS_CONTENT)) +
+      _REGISTER_ITEM).setResultsName('registers_items*')
+
+  _REGISTERS_HISTORY = (
+      pyparsing.Suppress(pyparsing.LineStart()) +
+      pyparsing.Literal('# Registers:') +
+      pyparsing.Suppress(pyparsing.LineEnd()) +
+      pyparsing.ZeroOrMore(_REGISTERS_ITEM))
+
+  _FILEMARKS_ITEM = pyparsing.Group(
+      pyparsing.Suppress(pyparsing.LineStart()) +  pyparsing.Literal('\'') +
+      pyparsing.Word(pyparsing.nums) +
+      pyparsing.Word(pyparsing.nums) +
+      pyparsing.Word(pyparsing.nums) +
+      pyparsing.restOfLine +
+      pyparsing.Suppress(pyparsing.LineEnd()) +
+      _FILEMARK_ITEM).setResultsName('filemarks_items*')
+
+  _FILEMARKS_HISTORY = (
+      pyparsing.Suppress(pyparsing.LineStart()) +
+      pyparsing.Literal('# File marks:') +
+      pyparsing.Suppress(pyparsing.LineEnd()) +
+      pyparsing.ZeroOrMore(_FILEMARKS_ITEM))
+
+  _JUMPLIST_ITEM = pyparsing.Group(
+      pyparsing.Suppress(pyparsing.LineStart()) +
+      pyparsing.Word('-\'') +
+      pyparsing.Word(pyparsing.nums) +
+      pyparsing.Word(pyparsing.nums) +
+      pyparsing.restOfLine +
+      pyparsing.Suppress(pyparsing.LineEnd()) +
+      _FILEMARK_ITEM).setResultsName('jumplist_items*')
+
+  _JUMPLIST_HISTORY = (
+      pyparsing.Suppress(pyparsing.LineStart()) +
+      pyparsing.Literal('# Jumplist (newest first):') +
+      pyparsing.Suppress(pyparsing.LineEnd()) +
+      pyparsing.ZeroOrMore(_JUMPLIST_ITEM))
+
+  _HISTORY_MARKS_HISTORY = (
+      pyparsing.Suppress(pyparsing.LineStart()) +
+      pyparsing.Literal('# History of marks within files (newest to oldest):') +
+      pyparsing.Suppress(pyparsing.LineEnd()))
+
+  VIMINFO_STRUCTURE = (
+      _PREAMBLE +
+      _HLSEARCH +
+      _SEARCH_PATTERN +
+      pyparsing.Optional(_SUBSTITUTE_SEARCH_PATTERN) +
+      pyparsing.Optional(_SUBSTITUTE_STRING) +
+      _COMMAND_LINE_HISTORY +
+      _SEARCH_STRING_HISTORY +
+      _EXPRESSION_HISTORY +
+      _INPUT_LINE_HISTORY +
+      _DEBUG_LINE_HISTORY +
+      _REGISTERS_HISTORY +
+      _FILEMARKS_HISTORY +
+      _JUMPLIST_HISTORY +
+      _HISTORY_MARKS_HISTORY
+  ).parseWithTabs()
+
+  def __init__(self, file_data):
+    self.parsed_data = self.VIMINFO_STRUCTURE.parse_string(file_data)
+
+  def CommandLineHistory(self):
+    """Returns a list of command line history items."""
+    items = []
+    for i, item in enumerate(self.parsed_data.get('command_line_items', [])):
```

Comment:
style nit: per style guide don't use one letter variable names. also this might be better solved with list comprehension.

Reply:
I fixed the one letter variable name, however, I feel list comprehension in these cases make it a bit harder to read due to the enumerate and the dictionary item that gets created is not simple.  For now I have left it as is but happy to change.